applications/FluidDynamicsApplication/custom_constitutive/newtonian_2d_law.cpp

```diff
@@ -79,6 +84,82 @@ void  Newtonian2DLaw::CalculateMaterialResponseCauchy(Parameters& rValues)
     }
 }
 
+void Newtonian2DLaw::CalculateDerivative(
+    Parameters& rParameterValues,
+    const Variable<Vector>& rFunctionVariable,
+    const Variable<double>& rDerivativeVariable,
+    Vector& rOutput)
+{
+    KRATOS_TRY
+
+    if (rFunctionVariable == CAUCHY_STRESS_VECTOR) {
+        // computes derivatives of CAUCHY_STRESS_VECTOR
+        // resize and clear the output variable
+        if (rOutput.size() != 3) {
+            rOutput.resize(3);
+        }
+        rOutput.clear();
+
+        if (rDerivativeVariable.IsComponent() && rDerivativeVariable.GetSourceVariable() == STRAIN_RATE_2D) {
+            // compute derivatives w.r.t. STRAIN_RATE
+            const double mu = this->GetEffectiveViscosity(rParameterValues);
+            const double volumetric_part_derivative = ((rDerivativeVariable.GetComponentIndex() == 0) + (rDerivativeVariable.GetComponentIndex() == 1)) / 3.0;
```

Comment:
```suggestion
            const double volumetric_part_derivative = rDerivativeVariable.GetComponentIndex() == 0 || rDerivativeVariable.GetComponentIndex() == 1 ? 1.0 / 3.0 : 0.0;
```
I think it is more readable in this way

Reply:
done :)